molecular-dynamics/thermostats.md

```diff
@@ -1,165 +1,954 @@
+\tau = \frac{2 \pi}{\omega} \quad \rightarrow \quad \tau^2 = 2 \pi^2  \frac{Q}{g k_B T}
+$$
+
+In general, the factor $2 \pi^2$ is combined in the numerical value of $\tau$. Therefore, the relationship between the thermal mass and the characteristic time is:
+
+$$
+Q = g k_B T \tau^2
+$$
+
+$\tau$ is the strength of the thermostat as the reflection of thermal mass $Q$ depends on the frequencies of bond vibrations. For open-source packages like LAMMPS, HOOMD-blue, and GROMACS, $\tau$ for certain materials is suggested below.
+
+- Condensed phases / simple liquids: $\tau$ = 0.1 − 0.5 ps.
 
+- Biomolecules / soft matter: $\tau$ = 0.5 - 2.0 ps.
+
+- Crystalline solids (high-frequency phonons): $\tau$ = 0.05 - 0.2 ps.
+
+- Gas or small system: $\tau$ > 0.5 ps.
+
+Alternatively, a safe choice $\tau$ also relates to the timestep $\Delta t$. It is important to keep  longer than the fastest simulated atomic behaviors (ex., Bond stretching) to avoid system distortion. In general, a safe range is:
+
+$$
+\tau \approx (50 - 200) \Delta t
+$$
+
+Some packages allow direct implementation of $Q$, but it is important to check the unit conversion. For example, in LAMMPS (metal units),$k_B = 8.617333 \cdot 10^{-5} \,\mathrm{eV / K}$, so $Q$ needs to be converted into the unit of $\mathrm{eV \cdot ps^2}$.
+
+It is also a good idea to check the results after picking $Q$. Some common rules of thumb include:
+
+- The temperature trace should relax to the target with mild fluctuations (or equilibrate with an isokinetic thermostat or velocity rescaling if too slow).
+
+- Check the distribution of kinetic energy over temperature: $\frac{2 K}{k_B T} \in \chi_{g}^2$.
+
+- The velocity autocorrelation should not be overly damped.
+
+
+**Implementation of Nosé–Hoover thermostat**
+
+Before, we simulate the NVE ensemble with the Velocity Verlet algorithm, where the solution is based on the operator splitting of the Liouvillian:
+
+$$
+i \hat{L} = i \hat{L}_1 + i \hat{L}_2
+$$
+
+$$
+e^{i \hat{L} \Delta t} \approx e^{i \hat{L}_2 \frac{\Delta t}{2}} e^{i \hat{L}_1 \Delta} te^{i \hat{L}_2 \frac{\Delta t}{2}}
+$$
+
+where $i \hat{L}_1$ is the position update, and $i \hat{L}_2$ is the momentum update.
+
+The above setup ensures that the time integration preserves the time-reversibility and the sympleticity. Therefore, for the Nosé–Hoover thermostat, a symmetric Trotter–Suzuki decomposition, or Nosé–Hoover Velocity Verlet integrator, takes the same idea that updates the thermostat by a half step symmetrically around the original Velocity Verlet integrator:
+
+$$
+i \hat{L} = i \hat{L}_1 + i \hat{L}_2 + i \hat{L}_T
+$$
+
+$$
+e^{i \hat{L} \Delta t} \approx e^{i \hat{L}_T \frac{\Delta t}{2}} e^{i \hat{L}_2 \frac{\Delta t}{2}} e^{i \hat{L}_1 \Delta} te^{i \hat{L}_2 \frac{\Delta t}{2}} e^{i \hat{L}_T \frac{\Delta t}{2}} 
+$$
+
+Specifically, the operators are defined as:
+
+- Position update operator:
+$$
+i\hat{L}_1 = \frac{\mathbf{p}}{m} \cdot \frac{\partial}{\partial \mathbf{r}}
```

Comment:
the next three equations have $ too much for some reason... unsure how to fix. have a look.